airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/StagingFilenameGenerator.java

```diff
@@ -4,26 +4,31 @@
 
 package io.airbyte.integrations.destination.jdbc;
 
+import static io.airbyte.integrations.destination.jdbc.constants.GlobalDataSizeConstants.*;
+
 /**
  * The staging file is uploaded to cloud storage in multiple parts. This class keeps track of the
  * filename, and returns a new one when the old file has had enough parts.
  */
 public class StagingFilenameGenerator {
 
   private final String streamName;
-  private final int maxPartsPerFile;
 
   // the file suffix will change after the max number of file
   // parts have been generated for the current suffix;
   // its value starts from 0.
   private int currentFileSuffix = 0;
   // the number of parts that have been generated for the current
   // file suffix; its value range will be [1, maxPartsPerFile]
-  private int currentFileSuffixPartCount = 0;
+  private static int currentFileSuffixPartCount = 0;
+
+  // This variable is responsible to set the size of chunks size (In MB). After chunks created in
+  // S3 or GCS they will be uploaded to Snowflake or Redshift. These service have some limitations for the uploading file.
```

Comment:
what are the limitations? is there a link to them?

Reply:
Looks like this limitations are pretty huge. 

https://cloud.google.com/storage/quotas
https://aws.amazon.com/blogs/aws/amazon-s3-object-size-limit/

But the main motivation of chunks was to bypass timeout errors for the cloud-based storage services. It is how I understood.